### PR TITLE
Remove clusterid tag

### PR DESCRIPTION
**Pull request by spangenberg**
/cc @enxebre

https://github.com/openshift/cluster-api-provider-aws/pull/192

### CODE REVIEW

File: data/data/aws/master/main.tf [hunk 1]
```diff
@@ -99,10 +99,10 @@ resource "aws_instance" "master" {
     ignore_changes = ["ami"]
   }
 
-  # `clusterid` is required for machine-api
+  # `kubernetes.io/cluster/*` is required for machine-api
   tags = "${merge(map(
     "Name", "${var.cluster_id}-master-${count.index}",
-    "clusterid", "${var.cluster_id}",
+    "kubernetes.io/cluster/${var.cluster_id}", "owned",
```
Comment (abhinavdahiya) [inline data/data/aws/master/main.tf]: this is not required it is already provided through the root/main.tf

Comment (spangenberg) [reply data/data/aws/master/main.tf]: Good point thanks.

**Comment by wking** [comment]
This makes me so happy, thanks :).

**Comment by enxebre** [comment]
@spangenberg I might be wrong but this can't get merged before openshift/cluster-api-provider-aws#192, otherwise `getInstances` https://github.com/openshift/cluster-api-provider-aws/pull/192/files#diff-9d8c56b0090958d82aa8280b211d4bf8R69 won't find the machine, so in case of an event watched by the informer the actuator `Exist()` will return false and it will try to create a new one

**Comment by wking** [comment]
Rebased around #1620 with e31caa3 -> 24b5db5.  Looks good to me, but we're still waiting on openshift/cluster-api-provider-aws#192.

**Comment by wking** [comment]
openshift/cluster-api-provider-aws#192 landed.

/hold cancel
/retest

**Pull request merged**